tests/unit/acquisition/test_optimizer.py

```diff
@@ -38,10 +46,77 @@
         ),
     ],
 )
-def test_optimize(
-    search_space: Box | DiscreteSearchSpace,
+def test_optimize_discrete(
+    search_space: DiscreteSearchSpace,
     shift: list[float],
     expected_maximizer: list[list[float]],
 ) -> None:
-    maximizer = optimize(search_space, lambda x: 0.5 - quadratic(x - shift))
+    maximizer = optimize_discrete(search_space, _quadratic_sum(shift))
+    npt.assert_allclose(maximizer, expected_maximizer, rtol=1e-4)
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "search_space, shift, expected_maximizer",
+    [
+        (Box([-1], [2]), [1.0], [[1.0]]),  # 1D
+        (Box([-1, -2], [1.5, 2.5]), [0.3, -0.4], [[0.3, -0.4]]),  # 2D
+        (Box([-1, -2], [1.5, 2.5]), [1.0, 4], [[1.0, 2.5]]),  # 2D with maximum outside search space
+        (Box([-1, -2, 1], [1.5, 2.5, 1.5]), [0.3, -0.4, 0.5], [[0.3, -0.4, 1.0]]),  # 3D
+    ],
+)
+def test_optimize_continuous(
+    search_space: Box,
+    shift: list[float],
+    expected_maximizer: list[list[float]],
+) -> None:
+    maximizer = optimize_continuous(search_space, _quadratic_sum(shift))
     npt.assert_allclose(maximizer, expected_maximizer, rtol=2e-4)
+
+
+def _branin_sum(x: TensorType) -> TensorType:
+    return -tf.reduce_sum(branin(x), axis=-2)
+
+
+@random_seed
+@pytest.mark.parametrize("batch_size", [1, 2, 3, 5])
+@pytest.mark.parametrize(
+    "search_space, acquisition, maximizers",
+    [
+        (Box([0], [1]), _quadratic_sum([0.5]), ([[0.5, -0.5]])),
+        (Box([0, 0], [1, 1]), _branin_sum, BRANIN_MINIMIZERS),
+        (Box([0, 0, 0], [1, 1, 1]), _quadratic_sum([0.5, -0.5, 0.2]), ([[0.5, -0.5, 0.2]])),
+    ],
+)
+def test_optimize_batch(
+    search_space: Box, acquisition: AcquisitionFunction, maximizers: TensorType, batch_size: int
+) -> None:
+    batch_size_one_optimizer = optimize_continuous
+    batch_optimizer = batchify(batch_size_one_optimizer, batch_size)
+    points = batch_optimizer(search_space, acquisition)
+    assert points.shape == [batch_size] + search_space.lower.shape
+    for point in points:
+        tf.reduce_any(point == maximizers, axis=0)
```

Comment:
i missed an assert here. this might even need an extra `reduce_all`

Reply:
THis will not work, as we need to test "point is approx equal to maximizers". This cannot be done while working with a set of multiple possible maximisers. 

I propose remove the branin test case here and just using the npt.assert_allclose.

Reply:
i don't follow. This checks that all `point`s in `points` are maximizers

Reply:
so i'm confident this is possible, and the branin test tests more than we get from quadratic since branin is more interesting with multiple global minima

i'll approve and leave it with you to do if you reckon it's worth it